tree/tree/src/TTree.cxx

```diff
@@ -7307,7 +7307,9 @@ void TTree::Print(Option_t* option) const
       for (l=0;l<nl;l++) {
          leaf = (TLeaf *)const_cast<TTree*>(this)->GetListOfLeaves()->At(l);
          br   = leaf->GetBranch();
-         if (strchr(br->GetName(),'.')) {
+         auto bname = br->GetName();
```

Comment:
I wonder if `br == br->GetMother()` would work as well?

Reply:
Maybe: in what cases a branch has the same of its mother?

Reply:
`br == br->GetMother()` is not checking the name but only the pointer.  It is essentially checking 'is this branch its own (top level) mother' which is 'true' only for the top level branches.

Reply:
I tried here: https://github.com/root-project/root/pull/17777